src/utils/src/arg_parser.rs

```diff
@@ -330,12 +328,8 @@ impl<'a> Arguments<'a> {
         while let Some(arg) = iter.next() {
             self.validate_arg(arg)?;
 
-            // If the `arg` argument is indeed an expected one, set the value provided by user
-            // if it's a valid one.
-            let argument = self
-                .args
-                .get_mut(&arg[ARG_PREFIX.len()..])
-                .ok_or_else(|| Error::UnexpectedArgument(arg[ARG_PREFIX.len()..].to_string()))?;
+            // Safe to unwrap as arg is validated beforehand
+            let argument = self.args.get_mut(&arg[ARG_PREFIX.len()..]).unwrap();
```

Comment:
I know it's safe to unwrap here but, in case we will change the logic in `validate_arg()`, the `ok_or_else()` approach is safer, so I would leave here as it was before.

Reply:
@lauralt done 👍